com.civicrm.hrjobroles/tests/phpunit/api/v3/HrJobRolesTest.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+require_once 'HrJobRolesTestBase.php';
+
+class api_v3_HrJobRolesTest extends HrJobRolesTestBase {
+
+
+  function setUp() {
+    parent::setUp();
+  }
+
+  function tearDown() {
+    parent::tearDown();
+  }
+
+  /**
+   * check with empty array
+   */
+  function testJobRoleCreateEmpty() {
+    $this->callAPIFailure('HrJobRoles', 'create', array());
+  }
+
+  /**
+   * check if required fields are not passed
+   */
+  function testJobRoleCreateWithoutJobContract() {
+    $params = array(
+      'title' => 'test role',
+    );
+    $this->callAPIFailure('HrJobRoles', 'create', $params);
+  }
+
+  /**
+   *  Test creating new role with valid parameters
+   */
+  function testJobRoleCreate() {
+    // create contact
+    $contactParams = array('first_name'=>'walter', 'last_name'=>'white');
+    $contactID = $this->individualCreate($contactParams);
+    // create contract
+    $contract = $this->createJobContract($contactID, date('Y-m-d', strtotime('-14 days')));
+
+    $params = ['job_contract_id' => $contract->id, 'title' => 'test role', 'sequential' => 1];
+    $result = $this->callAPISuccess('HrJobRoles', 'create', $params);
+    $this->assertEquals($result['values'][0]['job_contract_id'], $contract->id);
+    $this->assertEquals($result['values'][0]['title'], 'test role');
+  }
+
+  /**
+   *  Test creating new role with valid
+   * (location, region, department, level, cost center ) fields
+   *
+   */
+  function testJobRoleCreateWithValidOptionValues() {
+    // create contact
+    $contactParams = array('first_name'=>'walter', 'last_name'=>'white');
+    $contactID = $this->individualCreate($contactParams);
+    // create contract
+    $contract = $this->createJobContract($contactID, date('Y-m-d', strtotime('-14 days')));
+
+    // create option group and values
+    $this->createSampleOptionGroupsAndValues();
+
+    $params = [
+      'job_contract_id' => $contract->id,
+      'title' => 'test role',
+      'location' => 'amman',
+      'region' => 'south amman',
+      'department' => 'amman devs',
+      'level_type' => 'guru',
+      'cost_center' => 'abdali',
+      'sequential' => 1
+    ];
+    $result = $this->callAPISuccess('HrJobRoles', 'create', $params);
+    $this->assertEquals($result['values'][0]['location'], 'amman');
+    $this->assertEquals($result['values'][0]['region'], 'south amman');
+    $this->assertEquals($result['values'][0]['department'], 'amman devs');
+    $this->assertEquals($result['values'][0]['level_type'], 'guru');
+    $this->assertEquals($result['values'][0]['cost_center'], 'abdali');
+  }
+
+  /**
+   *  Test creating new role with Invalid
+   * (location, region, department, level, cost center ) fields
+   *
+   */
+  function testJobRoleCreateWithInvalidOptionValues() {
+    // create contact
```

Comment:
for example doing this :

```php
$params = [
 'job_contract_id' => $contract->id,
 'title' => 'test role2',
 'location' => 'soutsh ammansasdad',
     ];
$result = civicrm_api3('HrJobRoles', 'create', $params);
```
where location value  'soutsh ammansasdad' is not a valid value will through and error telling you that.

but doing this 
```php
CRM_Hrjobroles_BAO_HrJobRoles::create($params);
```
with the same parameters used in the api method will not throw any error and will pass and store wrong data in the database .


I really didn't tracked why this happen , but it appear there are some additional validation happen in the core API layer that do that.